src/opt.cpp

```diff
@@ -619,6 +620,16 @@ void ispc::Optimize(llvm::Module *module, int optLevel) {
             }
         }
 #endif
+        // On ARM NEON targets LoadStoreVectorizer works great to combine
+        // multiple loads/stores into a single vector load/store as in #2052.
+        // However because of LoadStoreVectorizer after vector loads,
+        // elements are often extracted and processed individually using scalar operations.
+        // To vectorize these scalar operations, SLP vectorizer is used.
+        // Enabling it on other targets may be beneficial but require extensive testing.
+        if (ISPCTargetIsNeon(g->target->getISPCTarget())) {
```

Comment:
You may want to introduce a dev switch to enable it on other platforms, for example undef `--opt=enable-slp` or something like that.

Though in this case you should be watching for pass numbers do not collide if someone enables it on Xe.

Reply:
I added two dev options to enable with SLP and LoadStore Vectorizers